### ISSUE TITLE
Add support for customising the timeframe over which DPS is tracked

### ISSUE THREAD

**Issue by yuiidev** [HunterPie/HunterPie#244]
Right now HunterPie tracks DPS by, I assume, `damage / time` however if the monster runs away, or I'm unable to land any big moves for a long time my DPS just tanks and stays tanked for the rest of the hunt, making the DPS graph essentially useless as it will always start high and inevitably go down and isn't representative of your performance at specific points in the hunt.

Ideally HunterPie would track damage in a (configurable) interval let's say every 5 seconds and plot a new datapoint on the graph with the DPS you have dealt in those 5 seconds. The graph will then be representative of your performance and actions throughout the hunt as opposed to a simple `damage / time` calculation.

**Comment by Haato3o** [comment]
From what I understood you want it to basically calculate the DPS from an interval, plot it on the graph once the interval is over, reset the DPS and calculate it again for the next interval and so on. 
But that creates another issue, the plots will look very weird in the graph, let's say you set it to an interval of 5 seconds and on the first interval you do `600 damage`, the graph will plot it as `102/s`, then you do `1000` damage in the next interval, the graph will plot it as `200/s`, then you do only `200 damage` in the next interval, that will be plotted at `40/s`, this will make the graph look weird full of spikes and gaps, if you don't hit the monster for 5 seconds, it would also plot `0/s`.

That situation would look kinda like the image below, and doing more damage after those intervals would make the graph spike again.
![image](https://user-images.githubusercontent.com/35552782/196010053-0291b431-e786-4c6d-baab-2689c8e8f149.png)

And it would still not fix the issue of your DPS being tanked because you stopped attacking the monster, the only thing that would actually fix that is if HunterPie kept track of when the player is in combat and subtracted it from the elapsed quest timer, but for that to work, it would need to keep track of every player in the party's combat state to not make your DPS look a lot higher than other people's, however, that solution would create empty gaps between plots if you leave the combat state for too long.

### DPS Calculation strategies

As of today, HunterPie has 3 different ways to calculate DPS and you can change that in your damage meter widget settings:

- **Relative to quest timer (Default):** It's the default `damage / elapsed_time` formula.
- **Relative to join:** It will also take in consideration the time you joined the quest, this is useful if you join SOSes a lot, the formula is `damage / (elapsed_time - time_you_joined)` 
- **Relative to first hit:** It will take in consideration the time you first hit the monster, the formula is `damage / (elapsed_time - time_you_hit)`
--- 

Also, keep in mind, DPS stands for **damage per seconds**, and that's literally what it is, it's your total damage over time, if you stop attacking then obviously that will impact your DPS, that's how DPS meters work.

Also, it not always start high and go down, it usually becomes stable after some time as you can see in the image below.
![image](https://user-images.githubusercontent.com/35552782/196010384-1252cb73-c18d-432e-9cf3-270a36b93ac7.png)

**Comment by Demozo** [comment]
Correct, however what you have outlined as an issue, is how graphs with a time component generally tend to work in my understanding. I'm also not suggesting replacing the existing solution, but rather add another mode as I'm sure the current functionality is fine for some/most people.

The issue of my DPS being tanked is more so because the total time continues to increase after quest commence, join or first hit while my total damage stays the same during a monster chase or just being unable to attack. I am more interested in seeing the "moment to moment" drops and spikes as opposed to a very averaged out graph.

**Comment by TheSwerik** [comment]
I had the same idea and would also love to have this dps option!

**Comment by Kyle-B** [comment]
I was confused by these graphs as well. I think the issue is that what's being graphed isn't the Damage Per Second, it's the Total Damage Per Second. Total DPS/Time doesn't naturally make sense as a graph because it becomes stable over time (it gets harder and harder to change as you get further into a fight). Total DPS only really makes sense as a metric.

I think that the graphs should either be Total Damage/Time, or it's derivative DPS/Time. Total Damage/Time will look very similar to the current Total DPS/Time graphs, but won't have the compression as you get further into the fight. Another benefit since DPS is the derivative of Total Damage/Time, it can easily be resampled to any time window for the recorded hunts as well. This is how every DPS tracker for MMOs works that I'm aware of, here's an example from the main FFXIV tracker:

https://www.fflogs.com/reports/HpY1XgtPBVkmjDh2#fight=4&type=damage-done

I can potentially take a look at changing this if you agree that this plan would make sense.

(As an aside, I do think encounter splitting is useful, but understand that would be more work to track)

**Comment by Haato3o** [comment]
Added in #577
